packages/front-end/components/MetricAnalysis/HistogramGraph.tsx

```diff
@@ -0,0 +1,260 @@
+import React, { FC, useEffect, useState } from "react";
+import { ParentSizeModern } from "@visx/responsive";
+import { Group } from "@visx/group";
+import { scaleLinear } from "@visx/scale";
+import { AxisBottom, AxisLeft } from "@visx/axis";
+import { Bar } from "@visx/shape";
+import {
+  TooltipWithBounds,
+  useTooltip,
+  useTooltipInPortal,
+} from "@visx/tooltip";
+import { ScaleLinear } from "d3-scale";
+import styles from "@/components/Metrics/DateGraph.module.scss";
+
+interface Datapoint {
+  start: number;
+  end: number;
+  units: number;
+}
+
+type TooltipData = { x: number; y: number; d: Datapoint };
+
+interface HistogramGraphProps {
+  data: Datapoint[];
+  formatter: (value: number, options?: Intl.NumberFormatOptions) => string;
+  height?: number;
+  margin?: [number, number, number, number];
+}
+
+function getTooltipDataFromDatapoint(
+  datapoint: Datapoint,
+  data: Datapoint[],
+  innerWidth: number,
+  yScale: ScaleLinear<unknown, unknown, never>
+) {
+  const index = data.indexOf(datapoint);
+  if (index === -1) {
+    return null;
+  }
+  const x =
+    (data.length > 0 ? (index + 0.5) / (data.length + 1) : 0) * innerWidth;
+  const y = (yScale(datapoint.units ?? 0) ?? 0) as number;
```

Comment:
can we use the `.undfined(0)` and get rid of the `?? 0` im pretty new to d3 graphs but from the docs it seems like that would work

Reply:
also at the moment Y could be any value other than `null` and `undefined`
again im not sure what values could come out of `yScale()`

Reply:
We had some weird bug in the past where yScale would return something nonsensical, so i'd prefer to leave this guard in. The inner ?? 0 is unnecessary though.